Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,221 @@ namespace xr
                     float rawPose[7]{};
                     ArHitResult_getHitPose(xrContext->Session, hitResult, tempPose);
                     ArPose_getPoseRaw(xrContext->Session, tempPose, rawPose);
-                    HitResult hitResult{};
-                    RawToPose(rawPose, hitResult.Pose);
+                    HitResult hitTestResult{};
+                    RawToPose(rawPose, hitTestResult.Pose);
 
-                    hitResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
-                    filteredResults.push_back(hitResult);
+                    hitTestResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
+                    filteredResults.push_back(hitTestResult);
                     frameTrackables.push_back(trackable);
                 }
             }
         }
 
+        // Converts an image bitmap to grayscale assumes an RGB8 or RGB8A image for use with ARCore.
+        void ConvertBitmapToGrayscale(
```

Comment:
Would love if someone had a better more generic way to create grayscale bitmaps,  This example is largely based on a modified version from the ARCore sample code adapted for both RGB8 and RGB8A, but am open to ideas to make this more flexible.